### PR TITLE
Add event_property_is and event_property_contains props to PushConditions

### PR DESCRIPTION
**Pull request by tusooa**
Preview: https://pr1673--matrix-spec-previews.netlify.app

### CODE REVIEW

**Comment by tusooa** [comment]
> Thanks for the contribution!
> 
> Does this relate to a particular MSC? If so, please could you link it?
> 
> Also, please be aware that, as per [CONTRIBUTING.rst](https://github.com/matrix-org/matrix-spec/blob/main/CONTRIBUTING.rst#sign-off), we require sign-off from a legally recognised name. If you'd prefer to keep this contribution anonymous, please follow the instructions there for private sign-off.

The creator of DCO clarified that requiring a "legal name" is not an intention. https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=d4563201f33a022fc0353033d9dfeb1606a88330

I recognize that Matrix may have different intentions on this matter, but I will NOT provide a "legal name" to anyone just for the sake of ME contributing code to others, even if it is done privately. Requiring me to provide "legal name" is a great way to ensure that I will NEVER contribute code to any matrix-org repositories again.

If desired, please close this one and all other PRs I have made against any matrix-org repo, and roll back https://github.com/matrix-org/matrix-spec/pull/1671 .

**Comment by joshsimmons** [comment]
@tusooa Thank you so much for your contribution and your feedback on the DCO!

The team here will follow up with you about this PR in the next couple of days, but I wanted to jump in as the Managing Director of the Foundation to speak to the policy. Coincidentally, I was just discussing the specifics of our policy with the team last week! We are broadly on the same page about the issues with real name/legal name policies and are looking at amending our policy accordingly :) Not clear on timeline yet – got a lot of plates we're spinning – but I thought you deserved to know.

**Comment by richvdh** [comment]
Hi @tusooa, thanks for your patience on this. I've checked in with our legal team and although our policy is currently still that we prefer to have a legal name, we're prepared to grant exceptions in cases where sharing a real name is a particular concern. And as @joshsimmmons already said, he and the legal team are working on making the DCO process more inclusive generally.

On that basis, your existing sign-off is sufficient, thank you!

**Comment by richvdh** [comment]
> For cross-linking: seems I missed this in #1464?

Right. So these properties were added in https://github.com/matrix-org/matrix-spec-proposals/pull/3758 and  https://github.com/matrix-org/matrix-spec-proposals/pull/3966.

**Review by richvdh — DISMISSED** [review]

File: data/api/client-server/definitions/push_condition.yaml [hunk 1]
```diff
@@ -43,5 +43,11 @@ properties:
       optionally prefixed by one of, ==, <, >, >= or <=. A prefix of < matches
       rooms where the member count is strictly less than the given number and
       so forth. If no prefix is present, this parameter defaults to ==.
+  value:
+    type: ["string", "integer", "boolean", "null"]
```
[the remark this answers is not in the document]
Comment (richvdh) [reply data/api/client-server/definitions/push_condition.yaml:47]: Wait. According to https://swagger.io/docs/specification/data-models/data-types/, this is not valid. "`type` as a list is not valid in OpenAPI (even though it is valid in JSON Schema)".

However, it is definitely a pattern we use elsewhere, such as [here](https://github.com/matrix-org/matrix-spec/blob/c335f2b9413a8ed331972ff499788c52182d1c0a/data/api/client-server/definitions/push_rule.yaml#L20).

@KitsuneRal or @zecakeh, as people who know a lot more about OpenAPI than me: can you comment on this? Should we be updating the `render-object-table` to support multi-valued `type`s? Or should we be using `oneOf` here?

[the remark this answers is not in the document]
Comment (zecakeh) [reply data/api/client-server/definitions/push_condition.yaml:47]: It was not valid in OpenAPI 3.0, it is valid in OpenAPI 3.1, which is fully compatible with JSON Schema.

From "[Migrating from OpenAPI 3.0 to 3.1.0](https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0)":

> In line with JSON Schema, the `type` keyword can now define multiple types for a schema with an array.

[the remark this answers is not in the document]
Comment (richvdh) [reply data/api/client-server/definitions/push_condition.yaml:47]: Aha, thanks for the clarification; that's very helpful.

In that case I guess we should merge this and file a separate issue to fix the template.

**Review by richvdh — APPROVED** [review]

**Comment by richvdh** [comment]
Thanks again for the contribution @tusooa !

File: data/api/client-server/definitions/push_condition.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (richvdh) [reply data/api/client-server/definitions/push_condition.yaml:47]: https://github.com/matrix-org/matrix-spec/issues/1675

**Pull request merged**